services/QuillLMS/client/app/bundles/PremiumHub/components/CanvasInstanceForm.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react'
+
+import { requestGet, requestPost } from '../../../modules/request';
+import CanvasInstanceSchoolsSection from './CanvasInstanceSchoolsSection';
+
+export const CREATE_CANVAS_INSTANCE_PATH = "/canvas_instances"
+export const SCHOOLS_WITH_SUBSCRIPTIONS_PATH = "/subscriptions/school_admin_subscriptions"
+
+type InputElement = React.ChangeEvent<HTMLInputElement>
+
+const CanvasInstanceForm = ({ passedSchools }) => {
+  const [url, setUrl] = React.useState('')
+  const [clientId, setClientId] = React.useState('')
+  const [clientSecret, setClientSecret] = React.useState('')
+  const [selectedSchoolIds, setSelectedSchoolIds] = React.useState([])
+  const [schoolsWithSubscriptions, setSchoolsWithSubscriptions] = React.useState(passedSchools || [])
+
+  const handleChangeUrl = (e: InputElement) => { setUrl(e.target.value) }
+  const handleChangeClientId = (e: InputElement) => { setClientId(e.target.value) }
+  const handleChangeClientSecret = (e: InputElement) => { setClientSecret(e.target.value) }
+
+  const handleSubmit = e => {
+    e.preventDefault()
+    if (!validInput) { return }
+
+    return requestPost(CREATE_CANVAS_INSTANCE_PATH, {
+      canvas_instance: {
+        url: url
+      },
+      canvas_config: {
+        client_id: clientId,
+        client_secret: clientSecret
+      },
+      canvas_instance_schools: { school_ids: selectedSchoolIds }
+    })
+  }
+
+  const isValidURL = (url: string): boolean => {
+    try {
+      const parsedURL = new URL(url);
+      return ['http:', 'https:'].includes(parsedURL.protocol);
+    } catch (error) {
+      return false;
+    }
+  }
+
+  React.useEffect(() => { getSchoolsWithSubscriptions() }, [])
+
+  const getSchoolsWithSubscriptions = () => {
+    requestGet(SCHOOLS_WITH_SUBSCRIPTIONS_PATH, (body) => {
+      setSchoolsWithSubscriptions(body.schools)
+    })
+  }
+
+  const validInput = (
+    url.length > 0
+    && isValidURL(url) && clientId.length > 0
+    && clientSecret.length > 0
+    && selectedSchoolIds.length > 0
+  )
+
+  const urlInput = (
+    <div>
+      <label>Url:</label>
+      <input
+        id='url'
+        onChange={handleChangeUrl}
+        type="text"
+        value={url}
+      />
+    </div>
```

Comment:
I don't think it really matters here, but I might have just used the [Backpack TextField component](https://www.quill.org/backpack#textFields) here (aka `Input`). It'll give you some automatic styling and accessibility stuff (for instance, technically here you'd want an `htmlFor=` on that label field because otherwise a screenreader can't tell which label corresponds to which input).

Reply:
Thank you, I forgot about the Backpack stuff.